test/integration/graphql_test.lua

```diff
@@ -779,3 +779,37 @@ function g.test_middleware()
         }
     )
 end
+
+g.test_default_values = function()
+    local server = cluster.main_server
+
+    server.net_box:eval([[
+        package.loaded['test'] = package.loaded['test'] or {}
+        package.loaded['test']['test_default_value'] = function(root, args)
+          return args.arg
+        end
+
+        local graphql = require('cartridge.graphql')
+        local types = require('cartridge.graphql.types')
+
+        graphql.add_callback({
+            name = 'test_default_value',
+            args = {
+                arg = types.string,
+            },
+            kind = types.string,
+            callback = 'test.test_default_value',
+        })
+    ]])
+
+    t.assert_equals(
+        server:graphql({
+            query = [[
+                query($arg: String = "default_value") {
+                    test_default_value(arg: $arg)
+                }
+            ]],
+        variables = {arg = box.NULL}}
```

Comment:
I believe, if there is an explicit `null` value, it should override default value. I suggest split this into 2 tests:

1. `variables = {}` - returns `default value`
2. `variables = {arg = box.NULL}` returns `nil`

Reply:
Also, consider a case, when argument is nonNull. Query should return an error if default value is provided, but variable has been set to `box.NULL`

Reply:
I agree with you - https://github.com/graphql/graphql-spec/pull/418 (seems it's covered in spec).

So, I'll try to fix it.

Reply:
@RunsFor 

I've updated my MR. Since that default arrays and objects are supported. And there is a difference between explicitly passed "null" and just omitted argument. However, I've noticed that nonNull arguments can't have default values -  https://github.com/tarantool/cartridge/blob/master/cartridge/graphql/rules.lua#L411